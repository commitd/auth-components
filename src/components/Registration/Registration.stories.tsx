import React from 'react'
import { Meta } from '@storybook/react'
import { Registration } from '.'
import { ThemeProvider } from '@committed/components'

export default {
  title: 'Components/Registration',
  component: Registration,
} as Meta

export const Default: React.FC = () => {
  return (
    <ThemeProvider>
      <Registration
        onSignUp={() => {}}
        email=""
        password=""
        confirmPassword=""
        onChangeEmail={() => {}}
        onChangePassword={() => {}}
        onChangeConfirmPassword={() => {}}
      />
    </ThemeProvider>
  )
}

export const ErrorMessage = () => (
  <ThemeProvider>
    <Registration
      onSignUp={() => {}}
      email=""
      password=""
      confirmPassword=""
      onChangeEmail={() => {}}
      onChangePassword={() => {}}
      onChangeConfirmPassword={() => {}}
      errorMessage="Username already exists"
    />
  </ThemeProvider>
)
