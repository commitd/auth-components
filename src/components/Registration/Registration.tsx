import React from 'react'
import {
  Container,
  Flex,
  Text,
  TextField,
  Button,
  Form,
  Box,
  Alert,
} from '@committed/components'

export interface RegistrationProps {
  onSignUp: () => void
  errorMessage?: string
  email: string
  password: string
  confirmPassword: string
  onChangeEmail: (e: React.ChangeEvent<HTMLInputElement>) => void
  onChangePassword: (e: React.ChangeEvent<HTMLInputElement>) => void
  onChangeConfirmPassword: (e: React.ChangeEvent<HTMLInputElement>) => void
  emailError?: boolean
  passwordError?: boolean
  submitting?: boolean
}

export const Registration: React.FC<RegistrationProps> = ({
  errorMessage,
  onSignUp,
  email,
  password,
  confirmPassword,
  onChangeEmail,
  onChangePassword,
  onChangeConfirmPassword,
  emailError,
  passwordError,
  submitting,
  children,
}) => {
  return (
    <Container maxWidth="sm">
      <Flex flexDirection="column" alignItems="center">
        {children}
        {errorMessage != null && errorMessage.length > 0 ? (
          <Box mb={3}>
            <Alert severity="error">
              <Text>{errorMessage}</Text>
            </Alert>
          </Box>
        ) : undefined}
        <Form width={1}>
          <Flex flexDirection="column">
            <TextField
              mb={3}
              required
              id="email"
              label="Email Address"
              name="email"
              autoComplete="email"
              autoFocus
              value={email}
              onChange={onChangeEmail}
              error={emailError}
            />
            <TextField
              mb={3}
              required
              name="password"
              label="Password"
              type="password"
              id="password"
              autoComplete="current-password"
              value={password}
              onChange={onChangePassword}
              error={passwordError}
            />
            <TextField
              mb={3}
              required
              name="confirm-password"
              label="Confirm Password"
              type="password"
              id="confirm-password"
              autoComplete="current-password"
              value={confirmPassword}
              onChange={onChangeConfirmPassword}
              error={passwordError}
            />
            <Button
              disabled={submitting}
              mb={1}
              fullWidth
              variant="contained"
              color="primary"
              onClick={onSignUp}
              type="submit"
            >
              Sign Up
            </Button>
          </Flex>
        </Form>
      </Flex>
    </Container>
  )
}
