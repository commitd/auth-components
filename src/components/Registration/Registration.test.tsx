import React from 'react'
import { render } from '../../setupTests'
import { Default } from './Registration.stories'

it('renders story', () => {
  const { asFragment } = render(<Default />)
  expect(asFragment()).toMatchSnapshot()
})
