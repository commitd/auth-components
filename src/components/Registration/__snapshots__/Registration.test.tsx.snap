// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders story 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root MuiContainer-maxWidthSm"
  >
    <div
      class="MuiBox-root MuiBox-root-3 Styled(MuiBox)-root-1 Styled(MuiBox)-root-2"
    >
      <form
        class="Component-root-6 Component-root-7 Component-root-4 Component-root-5"
      >
        <div
          class="MuiBox-root MuiBox-root-9 Styled(MuiBox)-root-1 Styled(MuiBox)-root-8"
        >
          <div
            class="MuiFormControl-root MuiTextField-root Component-root-10 Component-root-11 MuiFormControl-fullWidth"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-filled Mui-required Mui-required"
              data-shrink="true"
              for="email"
              id="email-label"
            >
              Email Address
              <span
                aria-hidden="true"
                class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
              >
                 *
              </span>
            </label>
            <div
              class="MuiInputBase-root MuiFilledInput-root makeStyles-root-12 MuiInputBase-fullWidth MuiInputBase-formControl"
            >
              <input
                aria-invalid="false"
                autocomplete="email"
                class="MuiInputBase-input MuiFilledInput-input"
                id="email"
                name="email"
                required=""
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="MuiFormControl-root MuiTextField-root Component-root-10 Component-root-17 MuiFormControl-fullWidth"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-filled Mui-required Mui-required"
              data-shrink="true"
              for="password"
              id="password-label"
            >
              Password
              <span
                aria-hidden="true"
                class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
              >
                 *
              </span>
            </label>
            <div
              class="MuiInputBase-root MuiFilledInput-root makeStyles-root-12 MuiInputBase-fullWidth MuiInputBase-formControl"
            >
              <input
                aria-invalid="false"
                autocomplete="current-password"
                class="MuiInputBase-input MuiFilledInput-input"
                id="password"
                name="password"
                required=""
                type="password"
                value=""
              />
            </div>
          </div>
          <div
            class="MuiFormControl-root MuiTextField-root Component-root-10 Component-root-18 MuiFormControl-fullWidth"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-filled Mui-required Mui-required"
              data-shrink="true"
              for="confirm-password"
              id="confirm-password-label"
            >
              Confirm Password
              <span
                aria-hidden="true"
                class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
              >
                 *
              </span>
            </label>
            <div
              class="MuiInputBase-root MuiFilledInput-root makeStyles-root-12 MuiInputBase-fullWidth MuiInputBase-formControl"
            >
              <input
                aria-invalid="false"
                autocomplete="current-password"
                class="MuiInputBase-input MuiFilledInput-input"
                id="confirm-password"
                name="confirm-password"
                required=""
                type="password"
                value=""
              />
            </div>
          </div>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained Component-root-19 Component-root-20 MuiButton-containedPrimary MuiButton-fullWidth"
            tabindex="0"
            type="submit"
          >
            <span
              class="MuiButton-label"
            >
              Sign Up
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
